tests/test_helper.tcl

```diff
@@ -839,6 +840,51 @@ proc is_a_slow_computer {} {
     expr {$elapsed > 200}
 }
 
+# The following functions and variables are used only when running large-memory
+# tests. We avoid defining when not running large-memory tests them because the 
+# global variable takes up lots of memory.
+if {$::large_memory} {
+    set ::str500 [string repeat x 500000000] ;# 500mb
```

Comment:
are you sure that when we reach here we already know that this variable is set?
i was thinking of adding these to util.tcl (not test_helper.tcl).
one of the disadvantages of the approach you took is that now it is allocated by every test client (we have 16 by default), and **also** by the host that spins them up (so 17).

i think a better way would be to declare an empty global variable in util.tcl, and when this method is executed, do a lazy init. so it'll only be allocated on first use.